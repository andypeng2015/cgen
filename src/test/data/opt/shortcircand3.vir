module foo

export function w $foo(l %a, l %i) {
@start:
  %k = slt.l %i, 3_l
  %f = flag.w %k
  %c = eq.w %f, 0_w
  br %c, @join, @cont
@cont:
  %k = sge.l %i, 0_l
  %f = flag.w %k
  jmp @join
@join:
  %c = eq.w %f, 0_w
  br %c, @bad, @ok
@ok:
  %s = mul.l %i, 4_l
  %a = add.l %a, %s
  %v = ld.w %a
  ret %v
@bad:
  ret 0xffffffff_w
}
