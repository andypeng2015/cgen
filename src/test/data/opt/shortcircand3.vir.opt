module foo

export function w $foo(l %a, l %i) {
@2:
  %1 = sge.l %i, 0x3_l ; @17
  br %1, @16(0xffffffff_w), @4
@4:
  %6 = sge.l %i, 0x0_l ; @22
  br %6, @11, @16(0xffffffff_w)
@11:
  %4 = lsl.l %i, 0x2_l ; @20
  %5 = add.l %a, %4 ; @21
  %v.1 = ld.w %5 ; @15
  jmp @16(%v.1)
@16(%0):
  ret %0
}
