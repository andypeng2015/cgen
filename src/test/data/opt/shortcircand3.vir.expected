module foo

export function w $foo(l %a, l %i) {
@0:
  %1 = sge.l %i, 0x3_l ; @15
  br %1, @14(0xffffffff_w), @2
@2:
  %6 = sge.l %i, 0x0_l ; @20
  br %6, @9, @14(0xffffffff_w)
@9:
  %4 = lsl.l %i, 0x2_l ; @18
  %5 = add.l %a, %4 ; @19
  %v.1 = ld.w %5 ; @13
  jmp @14(%v.1)
@14(%0):
  ret %0
}
